app/assets/javascripts/angular/directives/graph_chart.js

```diff
@@ -389,6 +384,15 @@ angular.module("Prometheus.directives").directive('graphChart', ["$location", "W
         return "<table class=\"labels_table\">" + labelRows.join("") + "</table>";
       }
 
+      function storeLegendState() {
+        if (!rsGraph) {
+          return;
+        }
+        rsGraph.series.forEach(function(s) {
```

Comment:
ah, right. i made this change because the previous implementation never removed the series, but now i see i just need to zero it out each refresh. good catch.